### ISSUE TITLE
[SP2019]Assemble Pi Power Solution

### ISSUE THREAD

**Issue by jcwilliamson** [sourcebots/tasks#10]

**Comment by jcwilliamson** [comment]
@trickeydan @Tyler-Ward Is this referring to the cables we made with the black putty stuff last year? or the power solution for the Pis on the arena? If it is the latter, would it be worth producing an issue to make power cables for the pi? If I recall correctly, the power cables we made last year had some issues.

**Comment by trickeydan** [comment]
> @trickeydan @Tyler-Ward Is this referring to the cables we made with the black putty stuff last year? or the power solution for the Pis on the arena? If it is the latter, would it be worth producing an issue to make power cables for the pi? If I recall correctly, the power cables we made last year had some issues.

There is another potential option: Building an adapter board that goes on top of the pi and lets you power them with a camcon. Would require appropriate protection circuitry ofc.

**Comment by jcwilliamson** [comment]
Making a board for it isn't a bad idea, particularly if we can find an IC that takes care of -ve voltage and overcurrent for us. In an ideal world, we could ship that with just the IC and some camcons on it, but we'd need to do some research into that.

**Comment by trickeydan** [comment]
> Making a board for it isn't a bad idea, particularly if we can find an IC that takes care of -ve voltage and overcurrent for us. In an ideal world, we could ship that with just the IC and some camcons on it, but we'd need to do some research into that.

Building a board is a good long-term solution as far as I can tell initially. I really hate those cables.

**Comment by Tyler-Ward** [comment]
The board should probably contain 
* A mini camcon. 
* An over voltage ad reverse polarity protection circuit
* The pi power input circuit to prevent back powering
* (optional) a couple of extra status LEDS e.g. power from our side.

The cases have a convenient slot on the outside of the case we can poke the board bit with cam-con through.

**Comment by kierdavis** [comment]
I like this idea.

> * An over voltage ad reverse polarity protection circuit

Maybe a good idea to stick a PPTC fuse in as well.

> * The pi power input circuit to prevent back powering

Not sure what "power input circuit" means here - am I right in thinking this refers to using a diode etc. to prevent the Pi from making power available on this board if it is also powered over USB?

> * a couple of extra status LEDS e.g. power from our side

A "nice-to-have" could be to include a few LEDs wired to GPIO pins for use by the robot API and/or the competitors' code e.g. to indicate that initialisation is completed / code is running / an exception has occurred / some expected hardware was not detected.

**Comment by jcwilliamson** [comment]
This has now been done, pending review. One issue is that the board is designed using symbols/footprints that are not open source, so there may be an issue with putting it on github. @kierdavis @Tyler-Ward do you have any advice for this?

**Comment by kierdavis** [comment]
It would be ideal if open source symbols/footprints could be used instead. If there are no suitable open source alternatives I guess we can just put the open source parts of the project on github and instruct users how to obtain the non-open-source parts in the README.

**Comment by Tyler-Ward** [comment]
The PCBs have been ordered, we now need to order the components @jcwilliamson can you provide a list of parts to order to build the boards and we can send that off to Son

**Comment by jcwilliamson** [comment]
I'll take care of it this evening.

**Comment by jcwilliamson** [comment]
@Tyler-Ward i've done it now, where would be the best place to put it?

**Comment by trickeydan** [comment]
> @Tyler-Ward i've done it now, where would be the best place to put it?

In the google drive, in the latest order sheet.

**Comment by jcwilliamson** [comment]
> How many boards are we making by the way? That way I can account for price
breaks.

At least 25

**Comment by RealOrangeOne** [comment]
Components need ordering

**Comment by trickeydan** [comment]
The components have been ordered.

**Comment by kierdavis** [comment]
I'm happy to help assemble these next weekend.

**Comment by joe9599** [comment]
I believe these need to be done in the reflow oven (for most components) so will need to happen midweek, @jcwilliamson can you confirm?

**Comment by jcwilliamson** [comment]
Doing them by hand would probably be possible, but the reflow oven would likely make it easier. Which reminds me, I still need to apply for B16 access again.

**Comment by joe9599** [comment]
If you can't do it during the week, then I should be able to help next weekend too, although I don't get there until Saturday evening

**Comment by jcwilliamson** [comment]
I'll apply for access during the week, it might be useful to have it at some point.

**Comment by kierdavis** [comment]
We have built:

* 18 fully working boards - these are now installed in the Raspberry Pis.
* one board which is electrically working but has a blockage in one of the header pins. This has been red-taped. It should be a simple job to fix.
* three boards which have all components except the header (because we were only able to order enough headers for the above 19 boards).

We have enough to ship for the summer school, plus a couple of spares, so I'm going to close this issue. I've opened #183 to track finishing assembly of these after the summer school.
